### ISSUE TITLE
Error when upgrading to uniforms@^1.16.0

### ISSUE THREAD

**Issue by vladejs** [vazco/uniforms#248]
I have this Container in my meteor app (created with `react-komposer`), which gets data from server and feed a `SelectField` to be used on a Form:
```
import { composeWithTracker } from 'react-komposer';

// dumb component
import { SelectField } from 'uniforms-bootstrap3';

const composition = (props, onData) => {
  if (Meteor.subscribe('users').ready()) {
    const options = Meteor.users
      .find()
      .fetch()
      .map(p => ({
        label: p.profile.fullname || p.username,
        value: p._id
      }));

    const placeholder = '-- Choose a user --';

    const onChange = value => {
      props.onChange(value);
    };

    onData(null, {
      ...props,
      options,
      onChange,
      placeholder
    });
  }
};

export default composeWithTracker(composition)(SelectField);
```
When inspecting the `props` on the Container, the method `onChange` is not present. So the following error was thrown:
```
Exception from Tracker recompute function:
meteor.js?hash=27829e9…:930 TypeError: props.onChange is not a function
    at Object.onChange (select_doctor_field_container.js:20)
    at _class.componentWillMount (modules.js?hash=a1970b9…:82956)
    at modules.js?hash=a1970b9…:39710
    at measureLifeCyclePerf (modules.js?hash=a1970b9…:39437)
    at ReactCompositeComponentWrapper.performInitialMount (modules.js?hash=a1970b9…:39709)
    at ReactCompositeComponentWrapper.mountComponent (modules.js?hash=a1970b9…:39620)
    at Object.mountComponent (modules.js?hash=a1970b9…:32577)
    at ReactCompositeComponentWrapper._updateRenderedComponent (modules.js?hash=a1970b9…:40127)
    at ReactCompositeComponentWrapper._performComponentUpdate (modules.js?hash=a1970b9…:40086)
    at ReactCompositeComponentWrapper.updateComponent (modules.js?hash=a1970b9…:40007)
```

Before upgrading, I was using `uniforms@^1.11.3` with `uniforms-bootstrap3@^1.11.3` with no errors

**Comment by radekmie** [comment]
I guess you were depending on some unspecified behaviour of the previous versions... Could you show an example of this field used in your form?

**Comment by vladejs** [comment]
I just pass it to the schema definition:

```
import SelectUsersFieldContainer from '/client/shared/forms/select_user_field_container'

const someSchema = new SimpleSchema({
  user: {
    type: String,
    uniforms: {
      component: SelectUsersFieldContainer,
      name: 'Some name'
  }
});
```

On older version, I expected an onChange prop on the Container, but now there are available only the props I pass in, in this case, `name`.
 
Oddly enough, by removing the `onChange` function on the container, the custom field works 0_o.
When the form gets rendered, I change the Select Field's values, and when submitted, the model gets the correct selected value from the custom Select Field. 

Somehow, the value is bypassing the container up to the form, which I don't understand why:
```
Form -> ContainerSelectField -> SelectField (when value is changed, the Form is aware)
```

**Comment by janowsiany** [comment]
`SelectUsersFieldContainer` is not a field created with [`connectField` helper](https://github.com/vazco/uniforms/blob/master/INTRODUCTION.md#guaranteed-props) so you should not expect onChange there.

Also, you should not be surprised that `SelectField` is still working as changes applied on this field are handled inside [SelectField](https://github.com/vazco/uniforms/blob/master/packages/uniforms-bootstrap3/src/SelectField.js) from [uniforms-bootstrap3](https://github.com/vazco/uniforms/tree/master/packages/uniforms-bootstrap3). This field was created with `connectField` helper and has `onChange` guaranteed.

**Comment by vladejs** [comment]
So, what's your recommendation? Move away the `onChange` on the container
('cause it works like before), or leave it as is and wrap it with
`connectField`?

2017-03-28 10:24 GMT-04:00 Jan Owsiany <notifications@github.com>:

**Comment by janowsiany** [comment]
@vladejs See **EDIT** in the comment above

**Comment by radekmie** [comment]
**TL;DR;** @janowsiany is right.

@vladejs  Why it was working? Because `AutoField` was unnecessarily passing all calculated props down which was both not needed and not documented. Why it's not working? Link above. What is my suggestion? Leave it as it is, because you don't have to alter the `onChange` _chain_. How it is working? Magic or as we call it React context.

I think everything is clear now.
